app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Admin::DeployGroupsController < ApplicationController
+  before_action :authorize_admin!
+  before_action :authorize_super_admin!, only: [ :create, :new, :update, :destroy ]
+  before_action :deploy_group, only: [:edit, :update, :destroy]
+
+  def index
+    @deploy_groups = DeployGroup.all
+  end
+
+  def new
+    @deploy_group = DeployGroup.new
+  end
+
+  def create
+    @deploy_group = DeployGroup.create(deploy_group_params)
+    if @deploy_group.persisted?
+      redirect_to action: 'index'
+    else
+      flash[:error] = "Failed to create deploy group: #{@deploy_group.errors.full_messages}"
+      render 'new'
+    end
+  end
+
+  def update
+    if deploy_group.update_attributes(deploy_group_params)
+      flash[:notice] = "Successfully saved deploy group: #{deploy_group.name}"
+      redirect_to action: 'index'
+    else
+      flash[:error] = "Failed to update deploy group: #{deploy_group.errors.full_messages}"
```

Comment:
I was mainly copying the edit stage. And @sandlerr I fixed the display of square brackets. Seems if you don't interpolate the string it'll create separate flashes for each msg.